test/db-migrations/mocha-setup.db-migrations.js

```diff
@@ -0,0 +1,52 @@
+global.assert = require('node:assert');
```

Comment:
Instead of making `assert` global, could we use Should.js instead? That's what we use in other Backend tests. Or Chai would be good too, since that's the direction we're planning to go (#1167). I think it's totally OK for things like `describeMigration()` in utils.js to use `assert`, but I think the tests themselves should use something else.

Reply:
:+1: I'll migrate the assertions to `chai` once other issues have been resolved.

Reply:
@matthew-white revisiting this, a few queries:

1. is this PR the appropriate place to add `chai` as a project dependency?
2. it helpful to introduce more uses of `should.js` when it's planned to be replaced?
3. is it really an issue using `assert` when it's a node built-in, and there is existing precedent for using it elsewhere in this codebase?  Specifically:

* `lib/model/migrations/20190520-01-add-form-versioning.js`
* `test/db-migrations/mocha-setup.js`
* `test/e2e/oidc/playwright-tests/src/utils.js`
* `test/e2e/standard/test.js`
* `test/integration/other/form-purging.js`
* `test/integration/other/submission-purging.js`
* `test/unit/data/entity.js`
* `test/unit/data/odata-filter.js`
* `test/unit/model/frames/entity.js`

Reply:
> is this PR the appropriate place to add `chai` as a project dependency?

Yes, feel free! Or you could it add to package.json in a separate PR if that feels better. We do plan to make that switch at some point.

> it helpful to introduce more uses of `should.js` when it's planned to be replaced?

You're definitely welcome to use Should.js, since that's what we currently use. Should.js and Chai are very similar, so it wouldn't take long to replace any Should.js assertions you add with Chai ones. In many/most cases, no change will be needed, e.g., `.should.equal()` in Should.js is `.should.equal()` in Chai.

> is it really an issue using `assert` when it's a node built-in, and there is existing precedent for using it elsewhere in this codebase?  Specifically:
> 
> * `lib/model/migrations/20190520-01-add-form-versioning.js`
> * `test/db-migrations/mocha-setup.js`
> * `test/e2e/oidc/playwright-tests/src/utils.js`
> * `test/e2e/standard/test.js`
> * `test/integration/other/form-purging.js`
> * `test/integration/other/submission-purging.js`
> * `test/unit/data/entity.js`
> * `test/unit/data/odata-filter.js`
> * `test/unit/model/frames/entity.js`

Our uses of `assert` have been limited, and with the exception of test/db-migrations/utils.js, I think we should try not to increase them. Almost all our tests use Should.js, in both Backend and Frontend.

test/integration/ and test/unit/ just use `assert` for `assert.throws()`. I don't think there's a particular reason for that. We could use `should.throws()` instead.

I don't mind `assert` being used in files that aren't modified often, like utility functions. I just want to avoid it in actual tests that developers are more likely to read or copy from. I'm also happy with custom assertion functions being named `assert*()`, e.g., `assertTableExists()`. I didn't realize that test/e2e/oidc/playwright-tests/src/utils.js used `assert`, but if those are utility functions, I don't think that's an issue.

On the other hand, maybe it'd be a good idea to replace uses of `assert` in test/e2e/standard/test.js, since those are actual tests. I see that that file was added recently. I don't feel like there's a pressing need to make that replacement though. I'm glad we have those tests.

For many migration tests, I suspect that there won't be a need for assertions beyond those made by the `assert*()` functions in utils.js. For example, test/db-migrations/20241008-01-add-user_preferences.spec.js doesn't use Should.js, but it also doesn't use `assert` directly.

Reply:
> I don't mind assert being used in files that aren't modified often, like utility functions. I just want to avoid it in actual tests that developers are more likely to read or copy from.

:+1: global removed.